### PR TITLE
cavs: use cavs version instead of platform names

### PR DESCRIPTION
**Pull request by abonislawski**
This patch will use cavs versions instead of platform names

### CODE REVIEW

File: src/include/sof/drivers/dmic.h [hunk 1]
```diff
@@ -32,16 +32,12 @@
 #define DMIC_HW_CONTROLLERS	2
 #define DMIC_HW_IOCLK		24000000
 #define DMIC_HW_FIFOS		2
-#endif
-
-#if CONFIG_SUECREEK
+#elif CONFIG_SUECREEK
```
Comment (mmaka1) [inline src/include/sof/drivers/dmic.h:35]: It looks incomplete. apl should be replaced with 1.5 and cnl with 1.8 (with special settings for sue inside).

File: src/platform/intel/cavs/platform.c [hunk 2]
```diff
@@ -145,7 +145,7 @@ EXT_MAN_PORT(
 ));
 #endif
 
-#if CONFIG_CANNONLAKE || CONFIG_ICELAKE || CONFIG_TIGERLAKE
+#if !CONFIG_SUECREEK && CAVS_VERSION >= CAVS_VERSION_1_8
```
Comment (mmaka1) [inline src/platform/intel/cavs/platform.c:148]: LPRO as the default clock not available for sue?

Comment (abonislawski) [reply src/platform/intel/cavs/platform.c]: (CONFIG_CANNONLAKE || CONFIG_ICELAKE || CONFIG_TIGERLAKE)
so probably no lpro

**Comment by lgirdwood** [comment]
SOFCI TEST

File: src/platform/intel/cavs/lib/clk.c [hunk 3]
```diff
@@ -54,7 +54,7 @@ static inline void select_cpu_clock_hw(int freq_idx, bool release_unused)
 		io_reg_write(SHIM_BASE + SHIM_CLKCTL,
 			     (io_reg_read(SHIM_BASE + SHIM_CLKCTL) &
 			      ~SHIM_CLKCTL_OSC_REQUEST_MASK) | enc);
-#if CONFIG_TIGERLAKE
+#if CASVS_VERSION == CAVS_VERSION_2_5
```
Comment (lgirdwood) [inline src/platform/intel/cavs/lib/clk.c:57]: typo here.

File: src/platform/intel/cavs/lib/clk.c [hunk 4]
```diff
@@ -31,7 +31,7 @@ static inline void select_cpu_clock_hw(int freq_idx, bool release_unused)
 	uint32_t enc = cpu_freq_enc[freq_idx];
 	uint32_t status_mask = cpu_freq_status_mask[freq_idx];
 
-#if CONFIG_TIGERLAKE
+#if CASVS_VERSION == CAVS_VERSION_2_5
```
Comment (lgirdwood) [inline src/platform/intel/cavs/lib/clk.c:34]: typo CASVS

**Review by lgirdwood — CHANGES_REQUESTED** [review]
Some typos

**Comment by paulstelian97** [comment]
Is this still needed?

**Pull request closed without merging**